### PR TITLE
Add Support for Python 3.12 and drop Python 2 Support

### PR DESCRIPTION
**Pull request by pbauer**
I'm not really sure about the main changes in the `Reloader` class. It seems to work as well as before (i.e. schema-changes are not applied but it works fine for for stuff like views and services). Maybe the whole thing could use a rewrite but I'm fine with what it does so far.

### CODE REVIEW

**Review by thet — APPROVED** [review]
LGTM

File: setup.py [hunk 1]
```diff
@@ -13,16 +13,18 @@
...
-        "Programming Language :: Python :: 2",
-        "Programming Language :: Python :: 2.7",
         "Programming Language :: Python :: 3",
         "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
+        "Programming Language :: Python :: 3.12",
```
Comment (ale-rt) [inline setup.py:26]: I would declare support for Python >= 3.8. 
But this is a really minor detail.

**Comment by ale-rt** [comment]
@jenkins-plone-org please run jobs

Of course I expect the Py 2.7 jobs to fail, let's see

File: plone/reload/xreload.py [hunk 2]
```diff
@@ -13,15 +13,16 @@
...
 import sys
 import types
 import inspect
 
 import six
-from six.moves import reload_module
+from importlib import reload as reload_module
 import zope.component
 
+# from zope.interface.interface import Specification
```
Comment (ale-rt) [inline plone/reload/xreload.py]:
```suggestion
```

**Review by ale-rt — APPROVED** [review]
LGTM

File: setup.py (hunk 1 above)
Comment (mauritsvanrees) [reply setup.py:26]: I will merge the PR, and then remove the 3.6 and 3.7 classifiers.

**Review by mauritsvanrees — APPROVED** [review]
Thanks!

I have created a branch 3.x and let coredev 5.2 use it.

**Comment by mauritsvanrees** [comment]
I have released `plone.reload` 4.0.0 and will update coredev 6.0 and 6.1.

**Pull request merged**